velox/common/caching/SsdFile.cpp

```diff
@@ -652,16 +669,21 @@ void SsdFile::initializeCheckpoint() {
   std::ifstream state(fileName_ + kCheckpointExtension);
   if (!state.is_open()) {
     hasCheckpoint = false;
+    ++stats_.errorsOpenCheckpoint;
     LOG(INFO) << "Starting shard " << shardId_ << " without checkpoint";
   }
   auto logPath = fileName_ + kLogExtension;
   evictLogFd_ = open(logPath.c_str(), O_CREAT | O_RDWR, S_IRUSR | S_IWUSR);
   if (evictLogFd_ < 0) {
-    // Failure to open the log at startup is a process terminating error.
-    LOG(ERROR) << "Could not open evict log " << logPath << " rc "
-               << evictLogFd_;
-    exit(1);
+    ++stats_.errorsOpenLog;
   }
+  // Failure to open the log at startup is a process terminating error.
+  VELOX_CHECK_GE(
+      evictLogFd_,
+      0,
+      "Could not open evict log {}, rc {}.",
+      logPath,
+      evictLogFd_);
```

Comment:
Shall we add 
recoverCheckpointErrors at l695 when readCheckpoint fails? Thanks!

Reply:
That's a great suggestion! I'm adding readCheckpointErrors, to be consistent with the naming of the rest.

Reply:
We usually use the term "recover" mean to rebuild the in-memory state from the last persistent checkpoint on disk.

Reply:
I'm not quite positive about my understanding, but the logic should be different here than recovering. Here we're not reading back the cache content according to the checkpoints. Here're we are reading the checkpoints and adjusting the metadata.

If you really think we're recovering cache state according to the checkpoints here, I will change.